Add "container.memory.usage.peak" metric

davidor opened the pull request:
### What does this PR do?

Documents the new `container.memory.usage.peak` metric.
It was introduced here: https://github.com/DataDog/datadog-agent/pull/18716
https://github.com/DataDog/datadog-agent/pull/19615


### Motivation
<!-- What inspired you to submit this pull request? -->

### Additional Notes
<!-- Anything else we should know when reviewing? -->

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] [Changelog entries](https://datadoghq.dev/integrations-core/guidelines/pr/#changelog-entries) must be created for modifications to shipped code
- [ ] Add the `qa/skip-qa` label if the PR doesn't need to be tested during QA.

FlorentClarret reviewed (dismissed):
Added a milestone so it will be easier to find these PRs when 7.48.0 will be out

davidor commented:
I've updated the PR to use the new name `container.memory.usage.peak`: https://github.com/DataDog/datadog-agent/pull/19615
I've also rebased on top of the master branch.
